stdlib/map.mli

```diff
@@ -86,6 +86,16 @@ module type S =
        of [x] in [m] disappears.
        @before 4.03 Physical equality was not ensured. *)
 
+    val update: key -> ('a option -> 'a) -> 'a t -> 'a t
```

Comment:
I can see two semantics for f returning None:
1. do not add/update the binding (we are happy with the value already bound or the absence of binding)
2. remove the binding
I am unsure of what is the most natural choice. What do you have in mind with your proposal?